webapp/platform/client/src/client4.ts

```diff
@@ -4221,6 +4222,32 @@ export default class Client4 {
             {method: 'post', body: JSON.stringify(body)},
         )
     }
+
+    private validateClient4URL(url?: string) {
+        const error = new ClientError(this.getUrl(), {
+            message: 'Invalid url.',
+            url,
+        })
+        if (!url) {
+            throw error;
+        }
+    
+        const path = url.split('?')[0].toLowerCase();
```

Comment:
Does this return `undefined` if there is no `?` ? Curious how this works with the test cases above

Reply:
[String.split](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split) should always return a array of strings with at least one element. So...
- `"?foo".split('?')`===`["", "foo"]`
- `"foo".split('?')`===`["foo"]`
- `"".split('?')`===`[""]`

The only way to have an empty list according to the documentation is doing `"".split("")` because a slight change on semantics. Nevertheles, that will never be done in any case. So... it is safe to assume that `url.split('?')[0]` is always defined and a string.

Reply:
Ah yes, makes sense. Thanks for the detailed explanation!